### PR TITLE
Bug 1486747 - Bouncer submission: Locations aren't added/checked if t…

### PR DESCRIPTION
**Pull request by JohanLorenzo**
…he product exists

### CODE REVIEW

File: bouncerscript/test/test_script.py [hunk 1]
```diff
@@ -40,28 +42,68 @@ def test_main(submission_context, fake_session):
...
+    mocker.patch.object(bscript, 'does_product_exist', new=return_false_async)
     with pytest.raises(ScriptWorkerTaskException):
         await bouncer_submission(submission_context)
 
 
+@pytest.mark.asyncio
+async def test_bouncer_submission_creates_locations_even_when_product_already_exists(
+    submission_context, mocker
+):
+    api_add_location_call_counter = (i for i in range(11))
```
Comment (tomprince) [inline bouncerscript/test/test_script.py]: You can use `itertools.count()`  instead, which doesn't require specifying an upper limit.

Comment (JohanLorenzo) [reply bouncerscript/test/test_script.py]: TIL! Done

File: bouncerscript/script.py [hunk 2]
```diff
@@ -28,36 +28,34 @@
...
-            context,
-            product_name=product_name,
-            add_locales=pr_config["options"]["add_locales"],
-            ssl_only=pr_config["options"]["ssl_only"]
-        )
-        log.info("Sanity check to ensure product has been successfully added...")
-        if not await does_product_exist(context, product_name):
-            raise ScriptWorkerTaskException("Bouncer entries are corrupt")
+            log.warning('Product "{}" already exists. Skipping...'.format(product_name))
+        else:
```
Comment (MihaiTabara) [inline bouncerscript/script.py]: I guess it's cleaner than using the continue statement 👍

File: bouncerscript/script.py [hunk 3]
```diff
@@ -28,36 +28,34 @@
...
-                err_msg = ("Corrupt submission entry for product {} platform "
-                           "{} path {}".format(product_name, platform, path))
-                raise ScriptWorkerTaskException(err_msg)
+            check_path_matches_destination(product_name, path)
         log.info("All submission entries look good before updating them!")
 
         log.info("Adding corresponding paths ...")
         for platform, path in pr_config["paths_per_bouncer_platform"].items():
-            await api_add_location(context, product_name, platform, path)
+            if await does_location_path_exist(context, product_name, path):
```
Comment (MihaiTabara) [inline bouncerscript/script.py]: Nice safeguarding here, thanks for adding this!

**Pull request merged**